### DISCUSSION TITLE
Tabbed Content - A great way to document tech-stuff

### DISCUSSION THREAD

**Discussion by zeekhuge** [outline/outline#5503] [Ideas]
Hi @tommoor 

It would be great to be able to add Tabs like this: 
![image](https://github.com/outline/outline/assets/8711218/098d9da0-b1c8-459b-b97d-96b52a77219e)

There are various scenarios where this is be super-duper handy. 

For reference 
https://bookdown.org/yihui/rmarkdown-cookbook/html-tabs.html
https://github.com/orgs/community/discussions/14964

**Comment by zeekhuge** [comment]
@tommoor : Would it be enough to just add a Node definition at `shared/editor/nodes/` ?

**Reply by zeekhuge** [reply]
Hey @tommoor ! Planning to work on this as well. 
The approach, as of now will be (based on a deep-dive into the code) :
To take the `Table` node as inspiration (even for commands).
1. Create Tab Node with
    - Another Node (similar to TableCell) to be used for the Tab name
        - This node will have content `<p>`
    - The content area will be `<p>`
    - Command to create Tab (with number of tabs as argument, similar to Table)
    - Command to add more Tabs (the way we do with Tables)
2. Style info in `component/**/Style.ts` (writing all this from memory, so don't remember exact address)
3. Expose these nodes in `node/index.ts`
4. Add these node in `menu/block.ts` with the correct setup.

Please, let me know if this is totally wrong. 

(Although, not totally sure when I'll start working on this)

**Reply by Huuuze** [reply]
You'll have a fan if you can make this happen. Tabbed content should be standard in wikis, IMO.

**Reply by tommoor** [reply]
It may be possible once #3000 is complete, but before that it's not possible as all content must have a lossless markdown representation.

**Reply by jonnymccullagh** [reply]
Any progress on this ?

**Reply by tommoor** [reply]
Nope, but if someone wants to work on adding it I'm happy to talk through how the implementation should work

**Comment by jellever** [comment]
+1 this would be a great feature.
